oneview_redfish_toolkit/services/computer_system_service.py

```diff
@@ -51,6 +53,50 @@ def get_storage_controller(server_profile_tmpl):
 
         return None
 
+    def power_off_server_hardware(self, sh_uuid_or_uri, on_compose=False):
+        """Power off a Oneview Server Hardware based on app configuration
+
+            Args:
+                sh_uuid_or_uri: the UUID or URI of Server Hardware to be
+                powered off
+                on_compose: if True gets configuration for the situation when
+                 want to Compose a System; if False gets configuration for the
+                 situation when want to Decompose a System
+
+            Returns:
+                True|False: if powering off action was applied to Oneview
+        """
+        if on_compose:
+            config_key = 'PowerOffServerOnCompose'
+        else:
+            config_key = 'PowerOffServerOnDecompose'
+
+        power_off = config.get_composition_settings()[config_key]
+
+        if power_off:
+            power_state = \
+                OneViewPowerOption.get_power_state_by_reset_type(power_off)
+
+            self.ov_client.server_hardware.update_power_state(power_state,
+                                                              sh_uuid_or_uri)
+
+            return True
+
+        return False
+
+    def validate_computer_system_resource_block_to_composition(self,
+                                                               system_block):
+        """Validates a Computer System Resource Block to be used in a composition
+
+            Args:
+                system_block: the Server Hardware that represents a Computer
+                System Resource Block
+        """
+        if system_block["serverProfileUri"]:
+            raise ValidationError(
+                "Computer System Resource Block already belongs to a "
+                "Composed System")
```

Comment:
Maybe we could return on the message the URL for the ComputerSystem existent.

Reply:
I agree with Victor to add the URL for Redfish ComputerSystem.

Reply:
done.